test/api/v3/integration/user/auth/POST-register_local.test.js

```diff
@@ -37,23 +37,94 @@ describe('POST /user/auth/local/register', () => {
+          username,
+          email,
+          password,
+          confirmPassword: password,
+        });
 
-      let user = await api.post('/user/auth/local/register', {
-        username,
-        email,
-        password,
-        confirmPassword: password,
+        expect(user.tags).to.have.a.lengthOf(7);
+        expect(user.tasksOrder.todos).to.have.a.lengthOf(1);
+        expect(user.tasksOrder.dailys).to.have.a.lengthOf(0);
+        expect(user.tasksOrder.rewards).to.have.a.lengthOf(0);
+        expect(user.tasksOrder.habits).to.have.a.lengthOf(0);
+      });
+
+      it('for Web', async () => {
+        api = requester(
+          null,
+          {'x-client': 'habitica-web'},
+        );
+        let username = generateRandomUserName();
+        let email = `${username}@example.com`;
+        let password = 'password';
+
+        let user = await api.post('/user/auth/local/register', {
+          username,
+          email,
+          password,
+          confirmPassword: password,
+        });
+
+        expect(user.tags).to.have.a.lengthOf(7);
+        expect(user.tasksOrder.todos).to.have.a.lengthOf(1);
+        expect(user.tasksOrder.dailys).to.have.a.lengthOf(0);
+        expect(user.tasksOrder.rewards).to.have.a.lengthOf(1);
+        expect(user.tasksOrder.habits).to.have.a.lengthOf(3);
+      });
+
+      it('for Android', async () => {
+        api = requester(
+          null,
+          {'x-client': 'habitica-android'},
+        );
+        let username = generateRandomUserName();
+        let email = `${username}@example.com`;
+        let password = 'password';
+
+        let user = await api.post('/user/auth/local/register', {
+          username,
+          email,
+          password,
+          confirmPassword: password,
+        });
+
+        expect(user.tags).to.have.a.lengthOf(7);
+        expect(user.tasksOrder.todos).to.have.a.lengthOf(2);
+        expect(user.tasksOrder.dailys).to.have.a.lengthOf(1);
+        expect(user.tasksOrder.rewards).to.have.a.lengthOf(1);
+        expect(user.tasksOrder.habits).to.have.a.lengthOf(2);
       });
 
-      expect(user.tags).to.have.a.lengthOf(7);
-      expect(user.tasksOrder.todos).to.have.a.lengthOf(1);
-      expect(user.tasksOrder.dailys).to.have.a.lengthOf(0);
-      expect(user.tasksOrder.rewards).to.have.a.lengthOf(0);
-      expect(user.tasksOrder.habits).to.have.a.lengthOf(0);
+      it('for iOS', async () => {
+        api = requester(
+          null,
+          {'x-client': 'habitica-ios'},
+        );
+        let username = generateRandomUserName();
+        let email = `${username}@example.com`;
+        let password = 'password';
+
+        let user = await api.post('/user/auth/local/register', {
+          username,
+          email,
+          password,
+          confirmPassword: password,
+        });
+
+        expect(user.tags).to.have.a.lengthOf(7);
+        expect(user.tasksOrder.todos).to.have.a.lengthOf(2);
```

Comment:
I know these lengthOf checks were in the previous code, but this is a structural check and is not concrete enough to verify the contents. So, in short, we should change these to verify the content of the tasks.